pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/TopicSchema.java

```diff
@@ -49,8 +53,13 @@ public class TopicSchema {
     private final Map<String, Schema<?>> cachedSchemas = new HashMap<>();
     private final PulsarClient client;
 
-    public TopicSchema(PulsarClient client) {
+    private final ClassLoader functionsClassloader;
+
+    public TopicSchema(PulsarClient client, ClassLoader functionsClassloader) {
         this.client = client;
+        this.functionsClassloader = AccessController.doPrivileged(
```

Comment:
Why do we need `AccessController.doPrivileged ` here?

Reply:
Interestingly IntelliJ IDEA gives me a pretty strong warning that doPrivileged should be used while creating this URLClassLoader. Shall we omit this call and just create the URLClassLoader as it is? I saw other parts of code that made use of doPrivileged (perhaps it also makes static analysis to fail).

Reply:
This method (as well as `AccessController`) is deprecated to be removed since JDK 17. We should find other way or just omit the warning.